[EDU-4321] feat: add guide - Turso Starter Kit (#885)
aziontech/docs: src/content/docs/en/pages/guides/marketplace/templates/turso-starter-kit.mdx

---

## Getting the template

To start using this template, access [Azion Console](https://console.azion.com/) and click the **+ Create** button on the homepage.

This will open a modal where you can select **Templates** > **Turso Starter Kit** card.

---

## Setting up the template

On the template's page, click the **Settings** tab to open the configuration form and provide the information to configure your application. 

Fields identified with an asterisk are mandatory.

- **Application Name** *: the name of your edge application on Azion.
- **GitHub Personal Token** *: your GitHub personal token.
- **Turso Database LibSQL URL** *: the LibSQL URL to connect with your Turso database.
- **Turso Database Token** *: the authorization token to access your Turso database.

After completing all the information, click the **Next** button, located in the bottom-right corner. This will start the deployment process.

:::note
If you provide an invalid GitHub personal token, the creation of the edge application will be executed, but the system won't be able to create the new repository in your GitHub Account. If the modules' subscription is disabled, the deployment will fail.

> 

Reviewer: ```suggestion
```

This page shows information about the application and some options to continue your journey.

:::note
The link to the edge application allows you to see it on the browser. However, it takes a certain time to propagate and configure the application in Azion's edge locations. It may be necessary to wait a few minutes for the URL to be activated and for the application page to be effectively displayed in the browser.
:::

Now you can [manage and adjust the settings](#managing-the-template) through Azion Console.

### Key configurations

This template deploys a basic edge application with a domain to access the interface and an edge function including a CRUD query example. Additionally, it creates an instance and a Default Rule to execute the function, as well as sets new environment variables for your Turso credentials.

It also creates a new GitHub repository, including a GitHub Action to enable a [continuous development workflow](#continuous-development) and keep your application up to date.

---

## Managing your database

> You can use the Azion domain to access it through the browser, as explained in the [deployment section](#deploying-the-template), and interact with it through the static interface that was created.

Reviewer 2: ```suggestion
You can use the Azion domain to access the application through the browser, as explained in the [deployment section](#deploying-the-template), and interact with the database through the static interface that was created.
```

Esses its me deixaram um pouco confusa hihi

By using this interface, you're able to create, retrieve, update, and delete items in your posts database, which is now linked to this new application. To do so, the interface provides an input area and icon buttons to manage your data as required.

> All changes will be reflected on your application and your Turso database as well.

Reviewer 2: Um detalhe que talvez seja importante falar que agora lembrei é o delete. É muito raro que algo seja de fato deletado de uma database, geralmente o que se faz é mover a entrada pra uma outra db de deletados. Assim se, por exemplo, esses dados forem úteis no futuro ou a pessoa tenha deletado por engano, é possível resgatar eles dessa outra db. Se o CRUD for o método delete puro, é bom explicar que o delete é o delete real e não a remoção/movimentação do dado.

Reviewer 3: Normalmente oq ocorre é um soft delete, que é quando na mesma tabela tem o campo de data de exclusao. Mas ali como fala do turso, acredito que esse delete seja de fato um delete.

1. Add the Azion personal token to the *secrets*:
- Read [how to generate an Azion personal token](/en/documentation/products/guides/personal-tokens/) in the documentation.

```bash
    AZION_PERSONAL_TOKEN=<value>
```

2. Add the environments for use in the action workflow in the **main.yml** file, included in the **.github/workflows** folder of your repository:

```yml
  - name: edge-...
    id: azion_edge
    ...
    with:
        ....
        azionPersonalToken: ${{ secrets.AZION_PERSONAL_TOKEN }}
        ....

```

3. Open a pull request to merge the changes to the main branch and start the automatic deployment.

Now, your project is ready to work with a continuous deployment workflow, updating instantly any changes in the application or the repository. 

### Adding a custom domain

The edge application created during the deployment has an assigned Azion domain to make it accessible through the browser. The domain has the following format: `xxxxxxxxxx.map.azionedge.net/`. However, you can add a custom domain for users to access your edge application through it.

import Button from '~/components/Button.astro';

> <Button href="/en/documentation/products/guides/configure-a-domain/" text="go to configuring a domain guide" variant="secondary"> 
> 
> </Button>

Reviewer 2: ```suggestion
<Button href="/en/documentation/products/guides/configure-a-domain/" text="go to configuring a domain guide" variant="secondary" />
```
